src/main.rs

```diff
@@ -228,22 +230,104 @@ fn hash_password(password: &str, pepper: &str, rounds: u32) -> String {
     let mut salt = [0u8; 16];
     let mut rng = OsRng::new().unwrap();
     rng.fill_bytes(&mut salt[..]);
-    let mut output = [0u8; 24];
 
     // ... See https://github.com/pyca/bcrypt/blob/fcebaa0db74dc822877128e57a79dcfda2a2dc4f/src/bcrypt/__init__.py#L66-L72
     // and https://github.com/DaGenix/rust-crypto/blob/cc1a5fde1ce957bd1a8a2e30169443cdb4780111/src/bcrypt.rs#L26
     let peppered_password = format!("{}{}", password, pepper);
-    let bcryptable_password = &peppered_password.as_bytes()[0..72];
+    let bcryptable_password: Vec<u8> = peppered_password.as_bytes().chunks(72).next().unwrap().to_vec();
+
+    info!("hashing with {}, {}", rounds, String::from_utf8(bcryptable_password.clone()).unwrap());
```

Comment:
obviously will be removed before merge, just trying to figure out why things don't match